src/main/java/minicraft/saveload/Load.java

```diff
@@ -437,99 +456,198 @@ private void loadUnlocks(String filename) {
+		if (worldVer.compareTo(new Version("2.0.7-dev4")) <= 0) {
+			for(int l = World.maxLevelDepth; l >= World.minLevelDepth; l--) {
+				LoadingDisplay.setMessage(Level.getDepthString(l));
+				int lvlidx = World.lvlIdx(l);
+				loadFromFile(location + filename + lvlidx + legacyExtension);
+				
+				int lvlw = Integer.parseInt(data.get(0));
+				int lvlh = Integer.parseInt(data.get(1));
+	
+				boolean hasSeed = worldVer.compareTo(new Version("2.0.7-dev2")) >= 0;
+				long seed = hasSeed ? Long.parseLong(data.get(2)) : 0;
+				Settings.set("size", lvlw);
+				
+				short[] tiles = new short[lvlw * lvlh];
+				JSONObject[] tdata = new JSONObject[lvlw * lvlh];
+				
+				for (int x = 0; x < lvlw; x++) {
+					for (int y = 0; y < lvlh; y++) {
+						int tileArrIdx = y + x * lvlw;
+						int tileidx = x + y * lvlw; // the tiles are saved with x outer loop, and y inner loop, meaning that the list reads down, then right one, rather than right, then down one.
+						String tilename = data.get(tileidx + (hasSeed ? 4 : 3));
+						if (worldVer.compareTo(new Version("1.9.4-dev6")) < 0) {
+							int tileID = Integer.parseInt(tilename); // they were id numbers, not names, at this point
+							if (Tiles.oldids.get(tileID) != null)
+								tilename = Tiles.oldids.get(tileID);
+							else {
+								System.out.println("Tile list doesn't contain tile " + tileID);
+								tilename = "grass";
+							}
 						}
-					}
-
-					if(tilename.equalsIgnoreCase("WOOL") && worldVer.compareTo(new Version("2.0.6-dev4")) < 0) {
-						switch (Integer.parseInt(extradata.get(tileidx))) {
-							case 1:
-								tilename = "Red Wool";
-								break;
-							case 2:
-								tilename = "Yellow Wool";
-								break;
-							case 3:
-								tilename = "Green Wool";
-								break;
-							case 4:
-								tilename = "Blue Wool";
-								break;
-							case 5:
-								tilename = "Black Wool";
-								break;
-							default:
-								tilename = "Wool";
+	
+						if(tilename.equalsIgnoreCase("WOOL") && worldVer.compareTo(new Version("2.0.6-dev4")) < 0) {
+							switch (Integer.parseInt(extradata.get(tileidx))) {
+								case 1:
+									tilename = "Red Wool";
+									break;
+								case 2:
+									tilename = "Yellow Wool";
+									break;
+								case 3:
+									tilename = "Green Wool";
+									break;
+								case 4:
+									tilename = "Blue Wool";
+									break;
+								case 5:
+									tilename = "Black Wool";
+									break;
+								default:
+									tilename = "Wool";
+							}
+						}
+	
+						if(l == World.minLevelDepth+1 && tilename.equalsIgnoreCase("LAPIS") && worldVer.compareTo(new Version("2.0.3-dev6")) < 0) {
+							if(Math.random() < 0.8) // don't replace *all* the lapis
+								tilename = "Gem Ore";
+						}
+						tiles[tileArrIdx] = Tiles.get(tilename).id;
+						JSONObject obj;
+						try {
+							obj = (JSONObject)Tiles.get(tilename).getClass().getMethod("getDefaultData").invoke(null, new Object[0]);
```

Comment:
Don't these calls have a negative performance impact?

Reply:
I think this is the only way to get the initial data of the tile given since the initial data can only be given by static method...I think it shouldn't be a method instead hmm